test_engine.py

```diff
@@ -0,0 +1,372 @@
+                    ResizeD(
+                        Keys.all(),
+                        spatial_size=[256, 256,128], 
+                        mode=("trilinear", "nearest", "nearest"), 
+                        allow_missing_keys = True,
+                    ),
+                    NormalizeIntensityD(Keys.IMAGE, channel_wise = True),
+                    ForegroundMaskD(
+                        Keys.LABEL,
+                        threshold = 0.5, 
+                        invert = True, 
+                        allow_missing_keys = True
+                    ),
+                    ToTensorD(Keys.all(), allow_missing_keys = True),
+                ]
+            )
+    engine.postprocessing_transforms = Compose([
+    ActivationsD(Keys.PRED,sigmoid = True),
+    AsDiscreteD(Keys.PRED,threshold = 0.5)
+    
+])
+#########################################################################################################################
+def test_load_data(engine):
+    """
+    Description: Tests the load_data method of the engine.
+
+    Parameters: engine(object) : An instance of the Engine class.
+
+    Returns: None
+    """
+
+    get_transforms(engine)
+    engine.load_data()
+
+    assert len(engine.train_dataloader) > 0
+    assert len(engine.test_dataloader) > 0
+    assert isinstance(engine.train_dataloader, MonaiLoader)
+    assert isinstance(engine.val_dataloader, MonaiLoader)
+    assert isinstance(engine.test_dataloader, MonaiLoader)
+
+
+#########################################################################################################################
+def test_save_checkpoint(engine,ckpt_path='Checkpoint'):
+    """
+    Description: Tests the save_checkpoint method of the engine.
+
+    Parameters:
+
+      engine(object) : An instance of the Engine class.
+
+      ckpt_path (str): path of the input directory.
+
+    Returns: None
+    """
+
+    engine.save_checkpoint(ckpt_path) 
+
+    assert os.path.exists(ckpt_path)
+    
+#########################################################################################################################
+def test_load_checkpoint(engine,checkpoint_path = 'Checkpoint'):
+    """
+    Description:  Tests the load_checkpoint method of the engine.
+
+    Parameters:
+       
+      engine(object) : An instance of the Engine class.
+
+      checkpoint_path (str): path of the input directory.
+
+    Returns: None
+    """
+    get_transforms(engine) 
+
+    init_weights = engine.network.state_dict()
+
+    engine.load_checkpoint(checkpoint_path)
+
+    # Get loaded weights
+    loaded_weights = engine.network.state_dict()
+
+    # Check that weights match
+    for i in init_weights.keys():
+        assert torch.allclose(init_weights[i], loaded_weights[i])
+#########################################################################################################################
+
+def test_fit(engine,
+        epochs = 2, 
```

Comment:
How do you pass these arguments to the test?

Reply:
i have fixed this